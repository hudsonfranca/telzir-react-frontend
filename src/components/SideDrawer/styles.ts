import styled from 'styled-components';

interface Props {
  isOpen: boolean;
}

export const Container = styled.div<Props>`
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background-color: var(--primary);
  transition: 0.8s ease-in-out;
  z-index: 20;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 2fr;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  transform: ${(props) =>
    props.isOpen ? 'translate3d(0,0,0)' : ' translate3d(-100%,0,0)'};

  div {
    margin: 3px 0 0 15px;
  }

  ul {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 100%;
    color: var(--white);
    font-size: 2rem;
    flex-direction: column;

    li {
      padding: 20px;
      width: 90%;
      display: flex;
      justify-content: center;
      align-items: center;

      :hover,
      :active {
        background-color: var(--secondary);
      }

      a {
        color: var(--white);
        transition: all 0.5s;
      }
    }
  }

  @media (min-width: 1000px) {
    display: none;
  }
`;

export default Container;
