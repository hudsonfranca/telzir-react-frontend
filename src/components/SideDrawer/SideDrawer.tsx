import React from 'react';

import { Link } from 'react-router-dom';
import { Container } from './styles';
import { useSideDrawer } from '../context/SideDrawerContext';
import HamburguerButton from '../HamburguerButton/HamburguerButton';

const SideDrawer: React.FC = () => {
  const { isOpen, setIsOpen } = useSideDrawer();
  return (
    <Container isOpen={isOpen} data-test="sidedrawerContainer">
      <HamburguerButton
        onClick={() => setIsOpen(!isOpen)}
        open={isOpen}
        data-test="hamburguer-button"
      />

      <ul data-test="menu">
        <li>
          <Link to="/" onClick={() => setIsOpen(!isOpen)}>
            Home
          </Link>
        </li>
        <li>
          <Link to="/about" onClick={() => setIsOpen(!isOpen)}>
            About
          </Link>
        </li>
        <li>
          <Link to="/contact" onClick={() => setIsOpen(!isOpen)}>
            Contact
          </Link>
        </li>
      </ul>
    </Container>
  );
};

export default SideDrawer;
