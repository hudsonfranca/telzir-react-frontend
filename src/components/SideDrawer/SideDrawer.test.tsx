import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { mount, ReactWrapper } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import SideDrawer from './SideDrawer';

let wraper: ReactWrapper;

beforeEach(() => {
  wraper = mount(
    <MemoryRouter>
      <SideDrawer />
    </MemoryRouter>
  );
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('<SideDrawer/> redering', () => {
  it('Should render without errors', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <SideDrawer />
        </MemoryRouter>
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  it('Should render 2 sidedrawerContainer', () => {
    expect(wraper.find(`[data-test='sidedrawerContainer']`).length).toEqual(2);
  });

  it('Should render one HamburguerButton', () => {
    expect(wraper.find(`[data-test='hamburguer-button']`).length).toBe(1);
  });

  it('Should render one Menu', () => {
    expect(wraper.find(`[data-test='menu']`).length).toBe(1);
  });
});
