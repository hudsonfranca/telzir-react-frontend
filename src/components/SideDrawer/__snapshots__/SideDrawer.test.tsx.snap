// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer/> redering Should render without errors 1`] = `
.c0 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background-color: var(--primary);
  -webkit-transition: 0.8s ease-in-out;
  transition: 0.8s ease-in-out;
  z-index: 20;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 2fr;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
}

.c0 div {
  margin: 3px 0 0 15px;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  color: var(--white);
  font-size: 2rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 ul li {
  padding: 20px;
  width: 90%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 ul li:hover,
.c0 ul li:active {
  background-color: var(--secondary);
}

.c0 ul li a {
  color: var(--white);
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c1 {
  width: 50px;
  height: 50px;
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 div {
  width: 100%;
  height: 3px;
  background-color: var(--white);
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
  border-radius: 10px;
}

.c1 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c1 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c1 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

@media (min-width:1000px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
  data-test="sidedrawerContainer"
>
  <div
    className="c1"
    data-test="hamburguerButtonContainer"
    onClick={[Function]}
  >
    <div
      data-test="hamburguerLine"
    />
    <div
      data-test="hamburguerLine"
    />
    <div
      data-test="hamburguerLine"
    />
  </div>
  <ul
    data-test="menu"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </li>
    <li>
      <a
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
    </li>
  </ul>
</div>
`;
